Solve `Test docs build` failing

e.g. https://github.com/fractal-analytics-platform/fractal-server/actions/runs/10453510531/job/28944117342?pr=1705

```console

2024-08-19T12:39:18.6678439Z INFO    -  Cleaning site directory
2024-08-19T12:39:18.6680193Z INFO    -  Building documentation to directory: /home/runner/work/fractal-server/fractal-server/site
2024-08-19T12:39:18.9613414Z INFO    -  Doc file 'changelog.md' contains an unrecognized relative link '../internals/version_upgrades/upgrade_1_2_5_to_1_3_0/', it was left as is. Did you mean 'internals/version_upgrades/upgrade_1_2_5_to_1_3_0.md'?
2024-08-19T12:39:19.0261073Z ERROR   -  Error reading page 'configuration.md': No module named 'griffe.collections'
2024-08-19T12:39:19.0262212Z Traceback (most recent call last):
2024-08-19T12:39:19.0271922Z   File "/opt/hostedtoolcache/Python/3.10.14/x64/bin/mkdocs", line 8, in <module>
2024-08-19T12:39:19.0273451Z     sys.exit(cli())
2024-08-19T12:39:19.0275046Z   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
2024-08-19T12:39:19.0276679Z     return self.main(*args, **kwargs)
2024-08-19T12:39:19.0278288Z   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/click/core.py", line 1078, in main
2024-08-19T12:39:19.0279717Z     rv = self.invoke(ctx)
2024-08-19T12:39:19.0281551Z   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
2024-08-19T12:39:19.0283112Z     return _process_result(sub_ctx.command.invoke(sub_ctx))
2024-08-19T12:39:19.0284823Z   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
2024-08-19T12:39:19.0286465Z     return ctx.invoke(self.callback, **ctx.params)
2024-08-19T12:39:19.0287633Z   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/click/core.py", line 783, in invoke
2024-08-19T12:39:19.0288538Z     return __callback(*args, **kwargs)
2024-08-19T12:39:19.0289476Z   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/mkdocs/__main__.py", line 286, in build_command
2024-08-19T12:39:19.0290344Z     build.build(cfg, dirty=not clean)
2024-08-19T12:39:19.0291331Z   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/mkdocs/commands/build.py", line 322, in build
2024-08-19T12:39:19.0292231Z     _populate_page(file.page, config, files, dirty)
2024-08-19T12:39:19.0293268Z   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/mkdocs/commands/build.py", line 175, in _populate_page
2024-08-19T12:39:19.0294175Z     page.render(config, files)
2024-08-19T12:39:19.0295462Z   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/mkdocs/structure/pages.py", line 271, in render
2024-08-19T12:39:19.0296315Z     self.content = md.convert(self.markdown)
2024-08-19T12:39:19.0297312Z   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/markdown/core.py", line 357, in convert
2024-08-19T12:39:19.0298153Z     root = self.parser.parseDocument(self.lines).getroot()
2024-08-19T12:39:19.0299181Z   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/markdown/blockparser.py", line 117, in parseDocument
2024-08-19T12:39:19.0300182Z     self.parseChunk(self.root, '\n'.join(lines))
2024-08-19T12:39:19.0301469Z   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/markdown/blockparser.py", line 136, in parseChunk
2024-08-19T12:39:19.0302363Z     self.parseBlocks(parent, text.split('\n\n'))
2024-08-19T12:39:19.0303460Z   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/markdown/blockparser.py", line 158, in parseBlocks
2024-08-19T12:39:19.0304333Z     if processor.run(parent, blocks) is not False:
2024-08-19T12:39:19.0305277Z   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/mkdocstrings/extension.py", line 124, in run
2024-08-19T12:39:19.0306317Z     html, handler, data = self._process_block(identifier, block, heading_level)
2024-08-19T12:39:19.0307401Z   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/mkdocstrings/extension.py", line 195, in _process_block
2024-08-19T12:39:19.0308375Z     handler = self._handlers.get_handler(handler_name, handler_config)
2024-08-19T12:39:19.0309546Z   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/mkdocstrings/handlers/base.py", line 459, in get_handler
2024-08-19T12:39:19.0310524Z     module = importlib.import_module(f"mkdocstrings_handlers.{name}")
2024-08-19T12:39:19.0311390Z   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/importlib/__init__.py", line 126, in import_module
2024-08-19T12:39:19.0415883Z     return _bootstrap._gcd_import(name[level:], package, level)
2024-08-19T12:39:19.0417427Z   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
2024-08-19T12:39:19.0419630Z   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
2024-08-19T12:39:19.0421043Z   File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
2024-08-19T12:39:19.0422267Z   File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
2024-08-19T12:39:19.0423543Z   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
2024-08-19T12:39:19.0424786Z   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
2024-08-19T12:39:19.0426989Z   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/mkdocstrings_handlers/python/__init__.py", line 3, in <module>
2024-08-19T12:39:19.0428909Z     from mkdocstrings_handlers.python.handler import get_handler
2024-08-19T12:39:19.0430907Z   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/mkdocstrings_handlers/python/handler.py", line 14, in <module>
2024-08-19T12:39:19.0432747Z     from griffe.collections import LinesCollection, ModulesCollection
2024-08-19T12:39:19.0434178Z ModuleNotFoundError: No module named 'griffe.collections'
2024-08-19T12:39:19.0860839Z ##[error]Process completed with exit code 1.
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -8,6 +8,8 @@
     * Remove `HAS_LOCAL_SBATCH` variable and related if-branches (\#1699).
 * Benchmarks:
     * Add `GET /auth/current-user/` to tested endpoints (\#1700).
+* Dependencies:
+    * Update `mkdocstrings` to `^0.25.2` (\#1707).
 
 # 2.3.7
 

--- a/docs/doc-requirements.txt
+++ b/docs/doc-requirements.txt
@@ -1,5 +1,5 @@
 mkdocs==1.5.3
-mkdocstrings[python]==0.24.3
+mkdocstrings[python]==0.25.2
 mkdocs-material==9.5.17
 mkdocs-gen-files==0.5.0
 mkdocs-literate-nav==0.6.1

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -71,7 +71,7 @@ jinja2 = "^3.1.3"
 [tool.poetry.group.docs.dependencies]
 
 mkdocs="1.5.3"
-mkdocstrings = { extras = ["python"], version = "0.24.3" }
+mkdocstrings = { extras = ["python"], version = "0.25.2" }
 mkdocs-material="9.5.17"
 mkdocs-gen-files="0.5.0"
 mkdocs-literate-nav="0.6.1"

